src/components/Chat/ChatMessage.tsx

```diff
@@ -32,21 +48,37 @@ class ChatMessage extends UIComponent<any, any> {
     variables: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
   }
 
-  static handledProps = ['as', 'children', 'className', 'content', 'mine', 'styles', 'variables']
+  static handledProps = [
+    'as',
+    'avatar',
+    'children',
+    'className',
+    'content',
+    'mine',
+    'styles',
+    'variables',
+  ]
 
   static defaultProps = {
     as: 'li',
   }
 
-  renderComponent({ ElementType, classes, rest }) {
-    const { children, content } = this.props
+  renderComponent({ ElementType, classes, rest, styles }) {
+    const { avatar, children, content, mine } = this.props
 
     return (
       <ElementType {...rest} className={classes.root}>
-        {childrenExist(children) ? children : content}
+        {!mine && this.renderAvatar(avatar, styles)}
+        <div className={classes.chatContent}>{childrenExist(children) ? children : content}</div>
+        {mine && this.renderAvatar(avatar, styles)}
       </ElementType>
     )
   }
+
+  private renderAvatar = (
+    avatarProps: IAvatarProps | undefined,
+    styles: IComponentPartStylesInput,
+  ) => avatarProps && <Avatar {...avatarProps} styles={{ root: styles.chatAvatar }} />
 }
```

Comment:
This is inconsistent with what we have in `Button` for icon.
I would suggest to add shorthand factory to `Avatar` and call `Avatar.create()` here.

Reply:
https://github.com/stardust-ui/react/blob/cb1f011993f8fd0fc07fa08e4c15a83cf19c0be9/src/components/Button/Button.tsx#L142-L147

Reply:
also, the fact that shorthand is properly used for the `avatar` prop should be tested - please, as an example consider to take a look on `Input-test.tsx` where `implementsShorthandProp` test method is used

Reply:
done both